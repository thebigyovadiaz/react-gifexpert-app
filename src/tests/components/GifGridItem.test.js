import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import { GifGridItem } from "../../components/GifGridItem";

describe("Pruebas Unitarias de <GifGridItem />", () => {
  const title = "Esto es titulo de la imagen";
  const url = "http://pruebas.testing.com";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("Debe hacer matchSnapshot del componente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Debe devolver el pÃ¡rrafo con el title", () => {
    const p = wrapper.find("p").text().trim();
    expect(p).toBe(title);
  });

  test("Debe devolver el alt del img igual a title y src igual a url", () => {
    const img = wrapper.find("img")
    const src = img.prop("src");
    const alt = img.prop("alt");

    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test("Debe tener en el div la clase animate__fadeInUpBig", () => {
    const div = wrapper.find("div").prop('className');
    const searchText = div.includes("animate__fadeInUpBig");
    expect(searchText).toBe(true);
  });
});
