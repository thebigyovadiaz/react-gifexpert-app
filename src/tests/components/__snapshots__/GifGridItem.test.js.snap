// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Unitarias de <GifGridItem /> Debe hacer matchSnapshot del componente 1`] = `
<div
  className="card animate__animated animate__fadeInUpBig"
>
  <img
    alt="Esto es titulo de la imagen"
    src="http://pruebas.testing.com"
  />
  <p>
    Esto es titulo de la imagen
  </p>
</div>
`;
